lib/Raven/ReprSerializer.php

```diff
@@ -29,7 +29,11 @@ protected function serializeValue($value)
         } elseif (is_integer($value) || is_float($value)) {
             return (string) $value;
         } elseif (is_object($value) || gettype($value) == 'object') {
-            return 'Object '.get_class($value);
+            $repr = 'Object ' . get_class($value);
+            if (method_exists($value, 'toSentry')) {
+                $repr .= ' ' . $value->toSentry();
```

Comment:
I'm thinking that if it provides toSentry maybe we just use that representation (and pass it back through the serializer). That means you could return an array of values or something similar.